R/nbastats.R

```diff
@@ -3,24 +3,72 @@
 #' @description
 #' The function filters the dataset further using the arguments provided and
 #' produces a tibble with summary statistics for a list of columns of a few players or teams.
-#' The function can only use one of the two filters - playerNames and teamNames. Hence, If playerNames
-#' are provided, teamNames are ignored.
+#' teamNames is a better argument to use  to get the summary stats for. The filters will be
+#' applied only when all is False, else the data is not filtered for any specific team, player
+#' or position.
 #'
-#' @param data tibble
+#' @param nba_data tibble
 #' @param columnNames character vector
 #' @param playerNames character vector
 #' @param teamNames character vector
+#' @param positions character vector
+#' @param all boolean
 #'
 #' @return A tibble
 #' @export
 #'
+#' @importFrom readr read_csv
+#' @import dplyr
+#'
 #' @examples
+#' \donttest{
+#' nba_data = read_csv(
+#' "https://raw.githubusercontent.com/kfoofw/nba_espn/master/0.data/NBA_reg_2018-2019.csv"
+#' )
 #' nbastats(nba_data, columnNames = c('GP', '3PM', 'FT%'),
 #'          playerNames = c('Stephen Curry', 'Paul George'))
 #' nbastats(nba_data, columnNames = c('GP', '3PM', 'FT%'),
 #'          teamNames = c('UTAH', 'PHX', 'DET'))
+#' }
 #'
-#'
-nbastats <- function(data, columnNames = c("all"),  playerNames = c("all"), teamNames = c("all")) {
-  print("nbastats function triggered!")
+nbastats <- function(nba_data, columnNames = c(),  playerNames = c(), teamNames = c(), positions = c(), all = FALSE) {
+
+  # Test inputs
+  if (!is.data.frame(nba_data)) {
+    stop('Data is not in correct format. A dataframe was expected by the function')
+  }
+  if (!is.logical(all)) {
+    stop('Function expects argument all to be a boolean.')
+
+  }
+
+  # Filter data
+  if (all == FALSE) {
+    if (length(playerNames) != 0) {
+      nba_data <- nba_data[nba_data$PLAYER %in% playerNames, ]
+    }
+
+    if (length(teamNames) != 0) {
+      nba_data <- nba_data[nba_data$Team %in% teamNames, ]
+    }
+
+    if (length(positions) != 0) {
+      nba_data <- nba_data[nba_data$POS %in% positions, ]
```

Comment:
Hi @vanandsh, as discussed verbally, the current `nbastats` function use may be complicated for the user. l

Let's modify the function to `nba_team_stats` where it no longer has a filter input based on player name. This new variant should be able to create descriptive statistics based on different teams. If any positions are specified as input, the filter should return descriptive stats for groupings of team AND positions.

Thanks for your hard work!

Reply:
Hi @kfoofw 

Thanks for the detailed review. I have made the changes requested for and tested it successfully on my local. Please review the latest changes.

Thanks
Anand